server/src/main/java/org/elasticsearch/search/fetch/FetchContext.java

```diff
@@ -0,0 +1,213 @@
+     */
+    public IndexFieldData<?> getForField(MappedFieldType fieldType) {
+        return searchContext.getForField(fieldType);
+    }
+
+    /**
+     * The original query
+     */
+    public Query query() {
+        return searchContext.query();
+    }
+
+    /**
+     * The original query with additional filters and named queries
+     */
+    public ParsedQuery parsedQuery() {
+        return searchContext.parsedQuery();
+    }
+
+    /**
+     * Any post-filters run as part of the search
+     */
+    public ParsedQuery parsedPostFilter() {
+        return searchContext.parsedPostFilter();
+    }
+
+    /**
+     * Configuration for fetching _source
+     */
+    public FetchSourceContext fetchSourceContext() {
+        return searchContext.fetchSourceContext();
+    }
+
+    /**
+     * Should the response include `explain` output
+     */
+    public boolean explain() {
+        return searchContext.explain();
+    }
+
+    /**
+     * The rescorers included in the original search, used for explain output
+     */
+    public List<RescoreContext> rescore() {
+        return searchContext.rescore();
+    }
+
+    /**
+     * Should the response include sequence number and primary term metadata
+     */
+    public boolean seqNoAndPrimaryTerm() {
+        return searchContext.seqNoAndPrimaryTerm();
+    }
+
+    /**
+     * Configuration for fetching docValues fields
+     */
+    public FetchDocValuesContext docValuesContext() {
+        FetchDocValuesContext dvContext = searchContext.docValuesContext();
+        if (searchContext.collapse() != null) {
+            // retrieve the `doc_value` associated with the collapse field
+            String name = searchContext.collapse().getFieldName();
+            if (dvContext == null) {
+                return new FetchDocValuesContext(Collections.singletonList(new FieldAndFormat(name, null)));
+            } else if (searchContext.docValuesContext().fields().stream().map(ff -> ff.field).anyMatch(name::equals) == false) {
+                dvContext.fields().add(new FieldAndFormat(name, null));
+            }
+        }
+        return dvContext;
+    }
+
+    /**
+     * Configuration for highlighting
+     */
+    public SearchHighlightContext highlight() {
+        return searchContext.highlight();
+    }
+
+    /**
+     * Should the response include scores, even if scores were not calculated in the original query
+     */
+    public boolean fetchScores() {
+        return searchContext.sort() != null && searchContext.trackScores();
+    }
+
+    /**
+     * Configuration for returning inner hits
+     */
+    public InnerHitsContext innerHits() {
+        return searchContext.innerHits();
+    }
+
+    /**
+     * Should the response include version metadata
+     */
+    public boolean version() {
+        // TODO version is loaded from docvalues, not stored fields, so why are we checking
+        // stored fields here?
+        return searchContext.version() &&
```

Comment:
It seems odd that disabling stored fields means that we don't return the version, but this does not apply to seqNo/primaryTerm?

Reply:
This does seem off! I think we could treat it as a bug and fix it in another PR.

Reply:
I opened https://github.com/elastic/elasticsearch/issues/62456